xtext/editorserver/Servlet.java

```diff
@@ -0,0 +1,228 @@
+import net.sourceforge.plantuml.FileFormatOption;
+import net.sourceforge.plantuml.SourceStringReader;
+
+/**
+ * Deploy this class into a servlet container to enable DSL-specific services.
+ */
+@WebServlet(name = "XtextServices", urlPatterns = "/xtext-service/*")
+public class DSLNAMEServlet extends XtextServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    DisposableRegistry disposableRegistry;
+
+    public void init() throws ServletException {
+        super.init();
+        Injector injector = new DSLNAMEWebSetup().createInjectorAndDoEMFRegistration();
+        this.disposableRegistry = injector.getInstance(DisposableRegistry.class);
+    }
+
+    private class ToXMIResponse {
+        public String output;
+    }
+
+    private class ErrorResponse {
+        public String error;
+    }
+
+    private class ToDiagramResponse {
+        public String diagram;
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String serviceID = req.getPathInfo().substring(1);
+        if ("to-xmi".equals(serviceID)) {
+            String sResult = exportXMI(req);
+
+            if (sResult != "") {
+                resp.setStatus(200);
+                resp.setContentType("application/json");
+
+                ToXMIResponse r = new ToXMIResponse();
+                r.output = sResult;
+
+                PrintWriter pw = resp.getWriter();
+
+                pw.print(new Gson().toJson(r));
+
+                pw.close();
+            } else {
+                resp.setStatus(200);
+                resp.setContentType("application/json");
+
+                ErrorResponse r = new ErrorResponse();
+                r.error = "Unable to generate XMI.";
+
+                PrintWriter pw = resp.getWriter();
+
+                pw.print(new Gson().toJson(r));
+
+                pw.close();
+            }
+        } else if ("to-diagram".equals(serviceID)) {
+            String sResult = exportDiagram(req);
+
+            if (sResult != "") {
+                resp.setStatus(200);
+                resp.setContentType("application/json");
+
+                ToDiagramResponse r = new ToDiagramResponse();
+                r.diagram = sResult;
+
+                PrintWriter pw = resp.getWriter();
+
+                pw.print(new Gson().toJson(r));
+
+                pw.close();
+            } else {
+                resp.setStatus(200);
+                resp.setContentType("application/json");
+
+                ErrorResponse r = new ErrorResponse();
+                r.error = "Unable to generate XMI.";
+
+                PrintWriter pw = resp.getWriter();
+
+                pw.print(new Gson().toJson(r));
+
+                pw.close();
+            }
+        } else {
+            super.doPost(req, resp);
+        }
+    }
+
+    private class RequestStructure {
+        public String input;
+    }
+
+    private String exportXMI(HttpServletRequest req)
```

Comment:
Add javadoc comments to document the methods etc.

Reply:
Very valid point.

Reply:
I've added some JavaDoc